tests/observers/docker_test.py

```diff
@@ -71,11 +71,38 @@ def test_docker_observer_labels_partial():
                 "nginx",
                 name="nginx-disco-partial",
                 labels={
-                    "agent.signalfx.com/config.80.extraDimensions": "{mydim: myvalue}",
+                    "agent.signalfx.com.config.80.extraDimensions": "{mydim: myvalue}",
                 }):
             assert wait_for(p(has_datapoint_with_dim, backend, "plugin", "nginx")), "Didn't get nginx datapoints"
             assert wait_for(p(has_datapoint_with_dim, backend, "mydim", "myvalue")), "Didn't get extra dimension"
         # Let nginx be removed by docker observer and collectd restart
         time.sleep(5)
         backend.datapoints.clear()
         assert ensure_always(lambda: not has_datapoint_with_dim(backend, "plugin", "nginx"), 10)
+
+
+def test_docker_observer_labels_multiple_monitors_per_port():
+    """
+    Test that we can configure multiple monitors per port using labels
+    """
+    with run_agent(dedent("""
+        observers:
+          - type: docker
+    """)) as [backend, get_output, _]:
+        with run_service(
+                "nginx",
+                labels={
+                    "agent.signalfx.com.monitorType.80": "collectd/nginx",
+                    "agent.signalfx.com.config.80.intervalSeconds": "1",
+                    "agent.signalfx.com.config.80.extraDimensions": "{app: nginx}",
+                    "agent.signalfx.com.monitorType.80-nginx2": "collectd/nginx",
+                    "agent.signalfx.com.config.80-nginx2.intervalSeconds": "1",
+                    "agent.signalfx.com.config.80-nginx2.extraDimensions": "{app: other}",
+                }):
+            assert wait_for(p(has_datapoint_with_dim, backend, "plugin", "nginx")), "Didn't get nginx datapoints"
+            assert wait_for(p(has_datapoint_with_dim, backend, "app", "nginx")), "Didn't get extra dims"
+            assert wait_for(p(has_datapoint_with_dim, backend, "app", "other")), "Didn't get extra dims"
+        # Let nginx be removed by docker observer and collectd restart
+        time.sleep(5)
+        backend.datapoints.clear()
+        assert ensure_always(lambda: not has_datapoint_with_dim(backend, "plugin", "nginx"), 10)
```

Comment:
Unrelated to your changes, but is this actually testing something or ensuring valid `run_service` teardown?  If the latter, should it be moved to the exit logic for that util instead of used in every test?

Reply:
It's ensuring that when you stop a service (nginx in this case) that the agent discovers that it has gone down and then stops monitoring it.  The way the test knows if the monitoring has stopped is by whether it gets datapoints or not.